src/Kestrel.Core/EndPointProtocols.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core
+{
+    [Flags]
+    public enum EndPointProtocols
```

Comment:
HttpProtocols

Reply:
Or just Protocols? In the specs they are ApplicationProtocols.

Reply:
No combined value?

Reply:
Flags enums usually have a default `None = 0` value.

Reply:
Yeah, I'd add `None` and `Http1AndHttp2` (don't call it `Both`, as that means we're screwed if we need to add more later ;)). `None` is a little weird because if you specify it, we should throw out an error, but it can be useful for initialization, comparison, null values, etc.

Reply:
I'm aware flags typically have a `None`, but it doesn't make sense here. Why add it just to check for it and throw? I don't see it being useful for the cases you pointed out (when would that happen?).

`Http1AndHttp2` is ugly but I'm willing to put it there. I don't see what's wrong with `|` though.

Reply:
The default value for enum fields is 0, None provides a name for that value. That way it's more obvious when you're working with an uninitialized field and you don't have to check for 0 in code.

Reply:
`Http1AndHttp2` is definitely ugly. I'm not entirely opposed to encouraging `|`, but most of our (the greater 'our', the one that includes .NET ;)) flags enums have common combinations as specific values and `Both` is limiting.

Regarding `None`, @Tratcher basically has the reasoning. Since you can cast any int to an enum and `0` is a common enough value for uninitialized enums/ints, it's very useful for debugging purposes to have a name for it. If you don't want an explicit error case, that's OK. I'm sure some kind of error would occur if neither flag was set. Specifying `None` can have the same error case as specifying `(HttpProtocol)42`, but having the name is a useful debugging error.